lib/src/file_context.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:io';
+
+import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/analysis/utilities.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:path/path.dart' as p;
+import 'package:source_span/source_span.dart';
+
+import 'patch.dart';
+
+/// A helper class for a file located at [path] that provides access to its
+/// contents and analyzed formats like [CompilationUnit] and [LibraryElement].
+class FileContext {
+  final AnalysisContextCollection _analysisContextCollection;
+
+  /// This file's path.
+  final String path;
+
+  FileContext(this.path, this._analysisContextCollection) {
+    if (!p.isAbsolute(path)) {
+      throw ArgumentError.value(path, 'path', 'must be absolute.');
+    }
+  }
+
+  /// A representation of this file that makes it easy to reference spans of
+  /// text, which is useful for the creation of [Patch]es.
+  SourceFile get sourceFile =>
+      _sourceFile ??= SourceFile.fromString(sourceText, url: Uri.file(path));
+  SourceFile _sourceFile;
+
+  /// The contents of this file.
+  String get sourceText => _contents ??= File(path).readAsStringSync();
+  String _contents;
+
+  /// Uses the analyzer to resolve and return the library result for this file,
+  /// which includes the [LibraryElement].
+  Future<ResolvedLibraryResult> getResolvedLibrary() =>
+      _analysisContextCollection
+          .contextFor(path)
+          .currentSession
+          .getResolvedLibrary(path);
+
+  /// Uses the analyzer to resolve and return the AST result for this file,
+  /// which includes the [CompilationUnit].
+  ///
+  /// If the fully resolved AST is not needed, use the much faster
+  /// [getUnresolvedUnit].
+  Future<ResolvedUnitResult> getResolvedUnit() => _analysisContextCollection
+      .contextFor(path)
+      .currentSession
+      .getResolvedUnit(path);
+
+  /// Returns the unresolved AST for this file.
+  ///
+  /// If the fully resolved AST is needed, use [getResolvedUnit].
+  CompilationUnit getUnresolvedUnit() =>
+      parseString(content: sourceText, path: path).unit;
+
+  /// Creates and returns a [Patch] for this file, where [startOffset] and
+  /// [endOffset] (or end of file, by default) define the span range and
+  /// [updatedText] is the text to write over that span.
+  Patch patch(String updatedText, int startOffset, [int endOffset]) =>
+      Patch(sourceFile, sourceFile.span(startOffset, endOffset), updatedText);
```

Comment:
For really simple suggestors, the most verbose part was creating the patch, so this util makes it even more terse.

Also note that I moved `updatedText` to the first param position, and made the same change in the `yieldPatch()` method on the two visitor suggestor mixins. I justified this breaking change because it allows `endOffset` to be optional, which mirrors the signature of `SourceFile.span()` where the `endOffset` will default to the end of the file if omitted.